### PR TITLE
Update hai missions.txt

### PR DESCRIPTION
**Pull request by jarekchr**
typos

NOTICE: Delete the sections that do not apply to your PR, and fill out the section that does.
(You can open a PR to add or improve a section, if you find them lacking!) 

----------------------
**Content (Artwork / Missions / Jobs)**

## Summary
2 typos in hai missions and some unneded commas removed

## PR Checklist
 - [x] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [ ] I uploaded the necessary image, blend, and texture assets here: {{insert link to assets}}
 - [ ] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}
  
  
-----------------------
**Bugfix:** This PR addresses issue #{{insert number}}

## Fix Details
{{add details}}

## Testing Done
{{describe how you tested that the fix doesn't introduce other issues}}

## Save File
This save file can be used to verify the bugfix. The bug will occur when using {{insert commit hash / version}}, and will not occur when using this branch's build.
{{attach a save file that can be used to verify your bugfix. It MUST have no plugin requirements}}

-----------------------
**Feature:** This PR implements the feature request detailed and discussed in issue #{{insert number}}

## Feature Details
N/A

## UI Screenshots
N/A

## Usage Examples
N/A

## Testing Done
N/A
## Performance Impact
N/A

### CODE REVIEW

File: data/hai/hai missions.txt [hunk 1]
```diff
@@ -510,15 +510,15 @@ mission "Unfettered returning home"
...
 				accept
 	
 	on visit
-		dialog `You look for the young Hai, but realize that he took a ride on one of your escorts! Better depart and wait for your escorts to arrive in this star system.`
+		dialog `You look for the young Hai but realize that he took a ride on one of your escorts! Better depart and wait for your escorts to arrive in this star system.`
 	on complete
 		payment 100000
 		dialog
-			`You need not have worried about the reception the Unfettered youth would receive here on Hai-home. Scarcely minutes after you hesitantly contact the Hai government, a happy crowd has gathered around your ship. When he steps out of the hatchway, he looks stunned at being welcomed so warmly. One of the Hai governors thanks you for transporting him, and pays you <payment>.`
+			`You need not have worried about the reception the Unfettered youth would receive here on Hai-home. Scarcely minutes after you hesitantly contact the Hai government, a happy crowd has gathered around your ship. When he steps out of the hatchway, he looks stunned at being welcomed so warmly. One of the Hai governors' thanks you for transporting him, and pays you <payment>.`
```
Comment (roadrunner56) [inline data/hai/hai missions.txt:521]: The apostrophe is not necessary here, as the word is not possessive.
```suggestion
			`You need not have worried about the reception the Unfettered youth would receive here on Hai-home. Scarcely minutes after you hesitantly contact the Hai government, a happy crowd has gathered around your ship. When he steps out of the hatchway, he looks stunned at being welcomed so warmly. One of the Hai governors thanks you for transporting him, and pays you <payment>.`
```

**Review by roadrunner56 — CHANGES_REQUESTED** [review]

**Review by Zitchas — COMMENTED** [review]
The spelling mistake corrections definitely look like good fixes to me. Good catch!

Someone else better versed in punctuation rules will need to check the various comma additions & deletions, though.

**Review by roadrunner56 — APPROVED** [review]

**Comment by Amazinite** [comment]
The two typo fixes look fine, but most if not all the changes to commas seem unnecessary. Relevant: https://github.com/endless-sky/endless-sky/pull/1190#issuecomment-223756706

**Comment by jarekchr** [comment]
You may be correct on my changes to commas as it seems the rule is to put it for example before "but" if there're 2 different clauses.
"You should put a comma before but only when but is connecting two independent clauses."

Reading again if the subject of the sentence is added to the 2nd clause (it's missing there) then my changes are incorrect.

**Comment by EjoThims** [comment]
Comma changes are indeed all correct, but some of them are pretty awkward sentences that might best be reworked entirely.

For several of them, this awkwardness is likely the origin of the bad comma placement in the first place.

**Comment by jarekchr** [comment]
I (or you) can remove all comma changes that I've made. Looking more deeply into other files and dialogues I think that there can be many such changes - perhaps you should consider this as some future item.

**Comment by Amazinite** [comment]
Well I was in the Hai file fixing something else so I decided to just pull in the typo changes from here instead of stripping out the comma changes.
If you think there are oddly worded sentences then you should suggest changes to them.

**Pull request closed without merging**